src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -871,6 +876,8 @@ public void InitializeComponent(IBuildComponentHost buildComponentHost)
                 _serviceState = LoggingServiceState.Initialized;
 
                 _buildEngineDataRouter = (buildComponentHost.GetComponent(BuildComponentType.BuildCheckManagerProvider) as IBuildCheckManagerProvider)?.BuildEngineDataRouter;
+
+                _buildCheckEnabled = buildComponentHost.BuildParameters!.IsBuildCheckEnabled;
```

Comment:
This `!` looks fishy (we're dereferencing `buildComponentHost.BuildParameters` above) and it builds successfully on my machine without it. Is it needed for you?